src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_SKCanvasElement.skia.cs

```diff
@@ -42,11 +42,98 @@ public async Task When_Clipped_Inside_ScrollViewer()
 		ImageAssert.DoesNotHaveColorInRectangle(bitmap, new Rectangle(0, 101, 400, 299), Microsoft.UI.Colors.Blue);
 	}
 
+	[TestMethod]
+	public async Task When_Loading_Not_Rendereing_Unnecessarily()
+	{
+		var renderInvalidatedCount = 0;
+
+		var SUT = new RedFillSKCanvasElement
+		{
+			Height = 400,
+			Width = 400
+		};
+
+		var border = new Border
+		{
+			BorderBrush = Microsoft.UI.Colors.Green,
+			Height = 400,
+			Child = new ScrollViewer
+			{
+				VerticalAlignment = VerticalAlignment.Top,
+				Height = 400,
+				Background = Microsoft.UI.Colors.Red,
+				Content = SUT
+			}
+		};
+
+		await UITestHelper.Load(border);
+
+		SUT.XamlRoot.RenderInvalidated += () =>
+		{
+			renderInvalidatedCount++;
+		};
+
+		// Wait a short time to ensure no extra invalidations occur
+		await Task.Delay(6000);
+
+		Assert.IsTrue(renderInvalidatedCount < 4, "RenderInvalidated should not be executed indefinitely.");
+	}
+
+
+	[TestMethod]
+	public async Task When_Invalidate_Called_MultipleTimes_DoesNot_Crash()
+	{
+		var SUT = new RedFillSKCanvasElement
+		{
+			Height = 200,
+			Width = 400
+		};
+
+		var blueCanvas = new BlueFillSKCanvasElement
+		{
+			Height = 200,
+			Width = 400
+		};
+
+		var stack = new StackPanel
+		{
+			VerticalAlignment = VerticalAlignment.Stretch,
+			HorizontalAlignment = HorizontalAlignment.Stretch,
+			Height = 500,
+		};
+
+		stack.Children.Add(SUT);
+		stack.Children.Add(blueCanvas);
+
+		var border = new Border
+		{
+			BorderBrush = Microsoft.UI.Colors.Green,
+			Height = 500,
+			Child = stack
+		};
+
+		await UITestHelper.Load(border);
+
+		var bitmap = await UITestHelper.ScreenShot(border);
+
+		ImageAssert.HasColorInRectangle(bitmap, new Rectangle(0, 0, 200, 200), Microsoft.UI.Colors.Red);
+		ImageAssert.HasColorInRectangle(bitmap, new Rectangle(0, 200, 200, 200), Microsoft.UI.Colors.Blue);
+	}
```

Comment:
This feels very similar to the preexisting `When_Clipped_Inside_ScrollViewer` test. and I'm not sure why we need a border here. Just a StackPanel would be enough I think. You should probably add something that invalidates measure as well.